kratos/includes/node.h

```diff
@@ -198,92 +168,85 @@ class Node : public Point, public Flags
+     */
+    Node(IndexType NewId, const double NewX, const double NewY, const double NewZ, 
+         VariablesList::Pointer pVariablesList, BlockType const * ThisData, SizeType NewQueueSize = 1);
 
-    /** Constructor using coordinates stored in given std::vector. Initialize
-    this point with the coordinates in the array. */
-    Node(IndexType NewId, std::vector<double> const&  rOtherCoordinates)
-        : BaseType(rOtherCoordinates)
-        , Flags()
-        , mNodalData(NewId)
-        , mDofs()
-        , mData()
-        , mInitialPosition()
-        , mNodeLock()
-    {
-        CreateSolutionStepData();
-    }
-
-    /// 3d with variables list and data constructor.
-    Node(IndexType NewId, double const& NewX, double const& NewY, double const& NewZ, VariablesList::Pointer  pVariablesList, BlockType const * ThisData, SizeType NewQueueSize = 1)
-        : BaseType(NewX, NewY, NewZ)
-        , Flags()
-        , mNodalData(NewId, pVariablesList,ThisData,NewQueueSize)
-        , mDofs()
-        , mData()
-        , mInitialPosition(NewX, NewY, NewZ)
-        , mNodeLock()
-    {
-    }
-
-    typename Node::Pointer Clone()
-    {
-        Node::Pointer p_new_node = Kratos::make_intrusive<Node >( this->Id(), (*this)[0], (*this)[1], (*this)[2]);
-        p_new_node->mNodalData = this->mNodalData;
-
-        Node::DofsContainerType& my_dofs = (this)->GetDofs();
-        for (typename DofsContainerType::const_iterator it_dof = my_dofs.begin(); it_dof != my_dofs.end(); it_dof++)
-        {
-            p_new_node->pAddDof(**it_dof);
-        }
-
-        p_new_node->mData = this->mData;
-        p_new_node->mInitialPosition = this->mInitialPosition;
-
-        p_new_node->Set(Flags(*this));
-
-        return p_new_node;
-    }
+    /**
+     * @brief Creates a clone of the current node.
+     * @details This function creates a new node with the same ID and coordinates as
+     * the current node, copies its nodal data, DOFs, and flags.
+     * @return Pointer to the cloned node.
+     */
+    typename Node::Pointer Clone();
 
-    /// Destructor.
-    ~Node() override
-    {
-        ClearSolutionStepsData();
-    }
+    /**
+     * @brief Destructor for the Node class.
+     * @details Clears any stored solution step data before destruction.
+     */
+    ~Node() override;
 
-    //*********************************************
-    //public API of intrusive_ptr
+    /**
+     * @brief Gets the reference count of the node.
+     * @details Public API of intrusive_ptr
+     * @return The current reference count.
+     */
     unsigned int use_count() const noexcept
     {
         return mReferenceCounter;
     }
-    //*********************************************
 
-    IndexType Id() const
-    {
-        return mNodalData.Id();
-    }
+    /**
+     * @brief Retrieves the node ID.
+     * @return The ID of the node.
+     */
+    IndexType Id() const;

```

Comment:
Some functions are so tiny and used so often that it's worth letting the compiler attempt to inline them. `Node::Id()` and friends are such functions and I often rely on them being inlined, so please don't move these to source files.

I recommend going through all functions and asking
1) how big is the binary footprint of this function?
2) how often is it invoked?

If the answers are 1) insignificant and 2) **very** often, just leave them in the header.

##

btw anything returning a ref/ptr to a member can be marked `noexcept`.